### PR TITLE
Store HTTPS listener information, add it to getCellInfo() (SYN-7632)

### PR DESCRIPTION
**Pull request by vEpiphyte**

### CODE REVIEW

File: synapse/lib/cell.py [hunk 1]
```diff
@@ -2822,6 +2823,8 @@ def reload():
 
             self.addReloadableSystem('https:certs', reload)
 
+        self.https_listeners.append({'host': lhost, 'port': lport})
```
Comment (MichaelSquires) [inline synapse/lib/cell.py:2826]: Maybe an optional `name=None` kwarg so the name can be put in this dictionary for a little extra context?

**Review by MichaelSquires — DISMISSED** [review]

File: synapse/lib/cell.py (hunk 1 above)
Comment (vEpiphyte) [reply synapse/lib/cell.py:2826]: what name would that be?

Comment (MichaelSquires) [reply synapse/lib/cell.py:2826]: I don't know. Some human context that someone can provide when calling `addHttpsPort()`. Something like `await addHttpsPort(1234, name='control')`? Never mind, it's a bad idea.

File: changes/191b3876553873b7a8ad0e2e61a43dd7.yaml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+---
+desc: Store the listening HTTPS host and port information in the base Cell. Expose
+  the listening https information in the getCellInfo API.
```
Comment (Cisphyx) [inline changes/191b3876553873b7a8ad0e2e61a43dd7.yaml:3]:
```suggestion
  the listening HTTPS information in the getCellInfo API.
```

File: changes/191b3876553873b7a8ad0e2e61a43dd7.yaml [hunk 3]
```diff
@@ -0,0 +1,6 @@
+---
+desc: Store the listening HTTPS host and port information in the base Cell. Expose
```
Comment (invisig0th) [inline changes/191b3876553873b7a8ad0e2e61a43dd7.yaml:2]: This delta does not "store" the HTTPS host/port ( nor should it ;) )

**Review by invisig0th — CHANGES_REQUESTED** [review]

File: changes/191b3876553873b7a8ad0e2e61a43dd7.yaml [hunk 4]
```diff
@@ -0,0 +1,7 @@
+---
+desc: |
+  The base Cell now captures runtime information for its listening HTTPS host and port information. This runtime
```
Comment (invisig0th) [inline changes/191b3876553873b7a8ad0e2e61a43dd7.yaml:3]: Mentioning class names seems a bit over-engineering-ish. KISS: "Update getCellInfo() API to include HTTPS listener host and port."

**Review by invisig0th — CHANGES_REQUESTED** [review]

File: synapse/tests/test_lib_cell.py [hunk 5]
```diff
@@ -503,6 +503,10 @@ async def test_cell_getinfo(self):
             cell.COMMIT = 'mycommit'
             cell.VERSION = (1, 2, 3)
             cell.VERSTRING = '1.2.3'
+
+            await cell.addHttpsPort(0)
+            await cell.addHttpsPort(0, host='127.0.0.1')
```
Comment (MichaelSquires) [inline synapse/tests/test_lib_cell.py:508]: This API returns the (host, port) tuple. You could capture it here and use it to validate the expected values later.

**Review by MichaelSquires — APPROVED** [review]

**Pull request merged**